### PR TITLE
Add product availability condition

### PR DESCRIPTION
**Pull request by rcmuniz1994**
#### What problem is this solving?

I created a condition to show or not show blocks based on product availability.

#### How to test it?

[https://dev249--unileverb2b.myvtex.com/](https://dev249--unileverb2b.myvtex.com/)

1. Access this workspace.
2. Click on "Iniciar sesión o registrate"
3. Click on "Crear una cuenta"
4. Register yourself on the store
5. Click on "Iniciar sesión o registrate"
6. Click on "Iniciar sesión"
7. Click on "Recibir código de acceso por e-mail"
8. Fill the input with your email.
9. Click on "Enviar"
10. Check your e-mail to get the code.
11. Fill the input with the received code.
12. Click on "Confirmar"
13. Access a available product (e.g. [link to an available product](https://dev249--unileverb2b.myvtex.com/mimosin-concentrado-suavizante-caricias-58-8lav/p?skuId=29)).
14. See that the buy button, price, sku selector and quantity selector are being shown.
15. Access a unavailable product (e.g. [link to an unavailable product](https://dev249--unileverb2b.myvtex.com/mimosin-concentrado-suavizante-moussel-58-8lav/p?skuId=28)).
16. See that only the sku select is being shown.

#### Screenshots or example usage:

An available product page:

available_product_page

An unavailable product page:

unavailable_product_page

### CODE REVIEW

File: react/ConditionLayoutProduct.tsx
Comment (kaisermann) [inline react/ConditionLayoutProduct.tsx:36]: What do you think of doing exactly like the `areAllVariationsSelected`?

We could replace `sellers` here with `isProductAvailable` and just return the ` isProductAvailable.length > 0`. The dev can control if they want it to be `false` or `true` via the `toBe` prop of the condition layout block.

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/ConditionLayoutProduct.tsx
Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx:36]: It's good idea. Thanks, @kaisermann . I'll try it.

Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx:36]: But...this way...will the product availability be verified to all of its sellers?

Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx]: @kaisermann I tried to implement your suggestion. Could you check it, pls?

File: react/ConditionLayoutProduct.tsx
Comment (kaisermann) [inline react/ConditionLayoutProduct.tsx:36]: This is closer! We should have no arguments here, so `undefined`
```suggestion
  isProductAvailable: undefined
```

File: react/ConditionLayoutProduct.tsx
Comment (kaisermann) [inline react/ConditionLayoutProduct.tsx:77]: And the availability logic should be here, not in the layout render method, so it's executed only when needed.

File: react/ConditionLayoutProduct.tsx
Comment (kaisermann) [inline react/ConditionLayoutProduct.tsx:106]: This can be moved to the `isProductAvailable` method like you were doing before! It returns if a product is available or not. If a dev wants to know if a product is NOT available, they can change their expectation by passing the `toBe: false` prop to the condition-layout component.

If you have questions or if I need to be clearer, just say it 😁

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/ConditionLayoutProduct.tsx
Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx]: @kaisermann Done. Could check it again, pls?

**Comment by rcmuniz1994** [comment]
@kaisermann I did what you asked. Could you review the code, please?

File: react/ConditionLayoutProduct.tsx
Comment (kaisermann) [inline react/ConditionLayoutProduct.tsx:84]: What do you think of it?

```suggestion
    const isAvailable = sellers?.some(
      (seller) => seller.commertialOffer.AvailableQuantity > 0
    )

    return isAvailable
```

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/ConditionLayoutProduct.tsx
Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx:84]: It's good idea. I'm goint to do it right now.

Comment (rcmuniz1994) [reply react/ConditionLayoutProduct.tsx]: @kaisermann It's done!

**Review by kaisermann — APPROVED** [review]

**Review by kaisermann — CHANGES_REQUESTED** [review]
Missing changelog 😁

**Comment by rcmuniz1994** [comment]
@kaisermann Could you check the changelog, pls?

**Review by kaisermann — APPROVED** [review]

**Comment by rcmuniz1994** [comment]
@kaisermann could you explain why I'm getting the following error from the Danger CI?

`Request failed [403]: https://api.github.com/repos/vtex-apps/condition-layout/issues/25/comments
Response: {
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/issues#create-an-issue-comment"
}`

It looks like a authentication error. Should I have which permission to get success on this check?

**Comment by kaisermann** [comment]
@rcmuniz1994 That's because danger doesn't have access to your fork. Not your fault 😉 We're just waiting for another review and we should be good to go.

**Review by klzns — APPROVED** [review]

**Comment by rcmuniz1994** [comment]
@kaisermann Now we have 2 approves. hihi :smile:

**Pull request merged**
